### PR TITLE
[TASK] Removed deprecated code from HtmlParser and RteHtmlParser

### PR DESCRIPTION
**Pull request by verdure-peuser**
Resolves: #483

https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/8.0/Breaking-72384-RemovedDeprecatedCodeFromHtmlParser.html

Provides 
RefactorRemovedMarkerMethodsFromHtmlParserRector (use MarkerBasedTemplateService for marker-methods, rename methods and remove some in HtmlParser)

RemoveRteHtmlParserEvalWriteFileRector (remove removed method in RteHtmlParserRector)

### CODE REVIEW

**Comment by tmotyl** [comment]
Hi @verdure-peuser 
Thanks for the PR!
Can you please describe what rectors are you providing  and link the PR with existing issues (there are issues for all breaking changes, so at last one has to match). Thanks!

**Comment by verdure-peuser** [comment]
Hi @tmotyl,

sorry, i've update the description. the issue was only linked in the commits, but they are hidden within the history. This Description is enough or do you need it somehow different ? Maybe we should define a Template for pull-requests to put in all required informations in a defined way ?

**Comment by tmotyl** [comment]
@verdure-peuser looks much better now. If you could also change PR title to sth like " Removed deprecated code from HtmlParser" that would be great.

**Comment by verdure-peuser** [comment]
@tmotyl done

**Comment by sabbelasichon** [comment]
@verdure-peuser Thank you very much. Beside a few minor readability things, it looks great. Much appreciated.

**Comment by verdure-peuser** [comment]
**Beside a few minor readability things,**, what are you suggesting ? Am open for improvements :)

**Comment by sabbelasichon** [comment]
@verdure-peuser Sorry, i did a review yesterday but didn´t submit it. Too tired.

File: src/Rector/v8/v0/RefactorRemovedMarkerMethodsFromHtmlParserRector.php [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    {
+        return [MethodCall::class, StaticCall::class];
+    }
+
+    /**
+     * @param StaticCall|MethodCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->isMethodStaticCallOrClassMethodObjectType($node, HtmlParser::class)) {
```
[the remark this answers is not in the document]
Comment (verdure-peuser) [reply src/Rector/v8/v0/RefactorRemovedMarkerMethodsFromHtmlParserRector.php:60]: please review again

File: src/Rector/v8/v0/RefactorRemovedMarkerMethodsFromHtmlParserRector.php [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            return null;
+        }
+        if (! $this->isNames($node->name, self::MOVED_METHODS_TO_MARKER_BASED_TEMPLATES)
+            && ! $this->isNames($node->name, self::REMOVED_METHODS)
+            && ! $this->isName($node->name, self::RENAMED_METHOD)) {
+            return null;
+        }
+
+        if ($this->isNames($node->name, self::MOVED_METHODS_TO_MARKER_BASED_TEMPLATES)) {
```
[the remark this answers is not in the document]
Comment (verdure-peuser) [reply src/Rector/v8/v0/RefactorRemovedMarkerMethodsFromHtmlParserRector.php]: please review again

**Pull request merged**